src/main/java/org/opentripplanner/routing/edgetype/StreetEdgeTraversal.java

```diff
@@ -129,6 +145,92 @@ public static StateEditor doTraverse(StreetEdge edge, State s0, RoutingRequest o
             time *= 0.001;
         }
 
+        StateEditor s1 = state.edit(edge);
+        s1.setBackMode(traverseMode);
+        s1.setBackWalkingBike(walkingBike);
+
+        ///////////////////////////////////
+        // NIH weighting preferences
+        OptionSet extraOptions = edge.getExtraOptionFields();
+        NumericFieldSet numericFieldSet = edge.getExtraNumericFields();
+        boolean isAudited = (extraOptions != null) || (numericFieldSet != null);
+
+        if (isAudited) {
+            LOG.info("Found audited edge {}: {}", edge.getName(), edge.getOsmId());
+
+            // prefer audited edges
+            weight *= 0.8;
+
+            // check for NIH routing params
+            OptionAttribute slope = extraOptions.getOption(NihOption.XSLOPE);
+            if ((slope != null) && (slope == XSlope.SLOPED)) {
+                s1.setHasUnevenSurfaces(); // mark state
+                if (!options.allowUnevenSurfaces) {
+                    LOG.info("Avoiding edge {}: {} due to slope", edge.getName(), edge.getOsmId());
+                    return null;
+                }
+            }
+
+
+            OptionAttribute rest = extraOptions.getOption(NihOption.REST);
+            if ((rest != null) && (rest != Rest.NONE_AVAILABLE)) {
+                s1.setPassesRestingPlaces(); // mark state
+                if (options.restingPlaces) {
+                    LOG.info("Preferring edge {}: {} with resting place {}", edge.getName(), edge.getOsmId(), rest.getLabel());
+                    weight *= 0.01;
+                    time *= 0.01;
+                }
+            }
+
+            if (options.crowding >= 0) {
+                double crowding = options.crowding;
+                LOG.info("Have crowding preference of {}", crowding);
+                // TODO: we don't have this data yet
+            }
+
+            // set weights generally based on NIH data
+            // TODO: why do we have both "niceness" and "pleasantness"?
+            // Does having these values mean we can ignore several of the other columns (traffic, disorder, etc?)
+            if (numericFieldSet != null) {
+                EnumMap<NihNumeric, Integer> numericExtras = numericFieldSet.getNumericValues();
+                int niceness = numericExtras.get(NihNumeric.NICENESS);
+                int pleasantness = numericExtras.get(NihNumeric.PLEASANTNESS);
+                int safety = numericExtras.get(NihNumeric.SAFE_SCORE);
+                // TODO: how to weight off of these values?
+            }
+
+            OptionAttribute curbRamp = extraOptions.getOption(NihOption.CURB_RAMP);
+            OptionAttribute surface = extraOptions.getOption(NihOption.SURFACE);
+            if (options.wheelchairAccessible) {
+                if ((curbRamp == CurbRamp.NO) || ((surface != null) && (surface != Surface.CONCRETE)) );
```

Comment:
Yeah, it would probably make sense to use this for the wheelchair edge banning.  I think I found some extra stuff needed for new input params that wasn't in the other branch.  Sorry to step on that; I thought the edge banning might be handled elsewhere.